.github/workflows/build.yml

```diff
@@ -88,3 +91,56 @@ jobs:
       with:
         name: instrumentation-test-results
         path: ./appmock/build/reports/androidTests/connected/**
+
+  analyze:
+    name: Analyze with SonarCloud
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Removed "needs" test, I think sonar analyser can run independently from any others jobs. Especially it will be faster to have this job as parallel to build & test.